### PR TITLE
[kjobctl] Add --priority flag to Slurm mode

### PR DESCRIPTION
**Pull request by IrvingMg**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Add `--priority` flag to support [Workload Priority Class](https://kueue.sigs.k8s.io/docs/concepts/workload_priority_class/).

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cmd/experimental/kjobctl/pkg/builder/slurm_builder.go [hunk 1]
```diff
@@ -75,6 +75,8 @@ const (
 
   echo "$replaced"
 }`
+
+	WorkloadPriorityClassLabel = "kueue.x-k8s.io/priority-class"
```
Comment (mbobrovskyi) [inline cmd/experimental/kjobctl/pkg/builder/slurm_builder.go:79]: I think we can use this constant:

https://github.com/kubernetes-sigs/kueue/blob/12fad5f72ba85f98d83b6557252f6b68de711e5b/pkg/controller/constants/constants.go#L35-L38

File: cmd/experimental/kjobctl/pkg/cmd/create/create_test.go [hunk 2]
```diff
@@ -1006,6 +1010,65 @@ error_path=$(unmask_filename "$SBATCH_ERROR")
...
+			gvks: []schema.GroupVersionKind{
+				{Group: "batch", Version: "v1", Kind: "Job"},
+				{Group: "", Version: "v1", Kind: "ConfigMap"},
+			},
+			wantLists: []runtime.Object{
+				&batchv1.JobList{
+					TypeMeta: metav1.TypeMeta{Kind: "JobList", APIVersion: "batch/v1"},
+					Items: []batchv1.Job{
+						*wrappers.MakeJob("", metav1.NamespaceDefault).
+							Label(builder.WorkloadPriorityClassLabel, "sample-priority").
```
Comment (mbobrovskyi) [inline cmd/experimental/kjobctl/pkg/cmd/create/create_test.go:1049]: Can we add a method on wrapper like `Profile()`?

File: cmd/experimental/kjobctl/pkg/builder/slurm_builder.go [hunk 3]
```diff
@@ -193,6 +195,18 @@ func (b *slurmBuilder) build(ctx context.Context) (runtime.Object, []runtime.Obj
...
+	if len(b.priority) != 0 {
+		// check that priority exists
+		_, err = b.kueueClientset.KueueV1beta1().WorkloadPriorityClasses().Get(ctx, b.priority, metav1.GetOptions{})
+		if err != nil {
+			return nil, nil, err
+		}
+
+		jobLabels := job.GetLabels()
+		jobLabels[WorkloadPriorityClassLabel] = b.priority
+		job.SetLabels(jobLabels)
```
Comment (mbobrovskyi) [inline cmd/experimental/kjobctl/pkg/builder/slurm_builder.go:207]:
```suggestion
		job.Labels[WorkloadPriorityClassLabel] = b.priority
```

File: cmd/experimental/kjobctl/pkg/builder/slurm_builder.go [hunk 4]
```diff
@@ -193,6 +195,18 @@ func (b *slurmBuilder) build(ctx context.Context) (runtime.Object, []runtime.Obj
 	job.ObjectMeta.GenerateName = ""
 	job.ObjectMeta.Name = objectName
 
+	if len(b.priority) != 0 {
```
Comment (mbobrovskyi) [inline cmd/experimental/kjobctl/pkg/builder/slurm_builder.go:198]: I also thinking about `--ignore-unknown-priority` to skip validation.

File: cmd/experimental/kjobctl/pkg/builder/slurm_builder.go [hunk 5]
```diff
@@ -193,6 +195,18 @@ func (b *slurmBuilder) build(ctx context.Context) (runtime.Object, []runtime.Obj
...
+		// check that priority exists
+		_, err = b.kueueClientset.KueueV1beta1().WorkloadPriorityClasses().Get(ctx, b.priority, metav1.GetOptions{})
+		if err != nil {
+			return nil, nil, err
+		}
```
Comment (mbobrovskyi) [inline cmd/experimental/kjobctl/pkg/builder/slurm_builder.go:203]: Can we move it to `validateGeneral()` function?

File: cmd/experimental/kjobctl/pkg/builder/slurm_builder.go (hunk 3 above)
Comment (IrvingMg) [reply cmd/experimental/kjobctl/pkg/builder/slurm_builder.go]: Done.

File: cmd/experimental/kjobctl/pkg/builder/slurm_builder.go (hunk 5 above)
Comment (IrvingMg) [reply cmd/experimental/kjobctl/pkg/builder/slurm_builder.go]: Done.

File: cmd/experimental/kjobctl/pkg/cmd/create/create_test.go (hunk 2 above)
Comment (IrvingMg) [reply cmd/experimental/kjobctl/pkg/cmd/create/create_test.go]: Done.

File: cmd/experimental/kjobctl/pkg/builder/slurm_builder.go (hunk 4 above)
Comment (IrvingMg) [reply cmd/experimental/kjobctl/pkg/builder/slurm_builder.go:208]: I added the flag with name `--skip-priority-validation`. 

WDYT?

Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/pkg/builder/slurm_builder.go:208]: Thanks!

File: cmd/experimental/kjobctl/pkg/builder/slurm_builder.go [hunk 6]
```diff
@@ -75,6 +76,8 @@ const (
 
   echo "$replaced"
 }`
+
+	WorkloadPriorityClassLabel = "kueue.x-k8s.io/priority-class"
```
Comment (mbobrovskyi) [inline cmd/experimental/kjobctl/pkg/builder/slurm_builder.go:80]: Can we revert it?

File: cmd/experimental/kjobctl/pkg/cmd/create/create_test.go [hunk 7]
```diff
@@ -1006,6 +1009,65 @@ error_path=$(unmask_filename "$SBATCH_ERROR")
 			},
 			wantOutPattern: `job\.batch\/.+ created\\nconfigmap\/.+ created`,
 		},
+		"should create slurm with --priority flag": {
```
Comment (mbobrovskyi) [inline cmd/experimental/kjobctl/pkg/cmd/create/create_test.go:1012]: Also could you add the test with `--skip-priority-validation`?

File: cmd/experimental/kjobctl/pkg/builder/slurm_builder.go (hunk 6 above)
Comment (IrvingMg) [reply cmd/experimental/kjobctl/pkg/builder/slurm_builder.go]: Done.

File: cmd/experimental/kjobctl/pkg/cmd/create/create_test.go [hunk 8]
```diff
@@ -1006,6 +1009,118 @@ error_path=$(unmask_filename "$SBATCH_ERROR")
...
+				cmpopts.IgnoreFields(corev1.LocalObjectReference{}, "Name"),
+				cmpopts.IgnoreFields(metav1.OwnerReference{}, "Name"),
+				cmpopts.IgnoreTypes([]corev1.EnvVar{}),
+				cmpopts.IgnoreTypes([]corev1.Volume{}),
+				cmpopts.IgnoreTypes([]corev1.VolumeMount{}),
+				cmpopts.IgnoreTypes(corev1.ConfigMapList{}),
+			},
+			wantOutPattern: `job\.batch\/.+ created\\nconfigmap\/.+ created`,
+		},
+		"should skip workload priority class validation": {
```
Comment (mbobrovskyi) [inline cmd/experimental/kjobctl/pkg/cmd/create/create_test.go:1071]:
```suggestion
		"should create slurm with --priority flag and skip workload priority class validation": {
```

Maybe like this?

**Review by mbobrovskyi — COMMENTED** [review]
I also thinking shouldn't we add this flag to another modes?

File: cmd/experimental/kjobctl/pkg/cmd/create/create_test.go (hunk 8 above)
Comment (IrvingMg) [reply cmd/experimental/kjobctl/pkg/cmd/create/create_test.go]: Done.

**Comment by IrvingMg** [comment]
> I'm also thinking shouldn't we add this label to another modes as well?

I think it might be useful for other modes as well.

@mwielgus WDYT?

**Comment by mbobrovskyi** [comment]
As I can see on the table https://kueue.sigs.k8s.io/docs/overview/#job-integrated-features almost all Jobs have `Workload Priority Class` integration except `Pod`. So it would be great to add to another modes (except `Interactive`) like we have `--localqueue`.

**Comment by mbobrovskyi** [comment]
> As I can see on the table https://kueue.sigs.k8s.io/docs/overview/#job-integrated-features almost all Jobs have `Workload Priority Class` integration except `Pod`. So it would be great to add to another modes (except `Interactive`) like we have `--localqueue`.

Let's make it in another PR.

/lgtm

**Review by mimowo — COMMENTED** [review]
/approve
Thanks!

**Review by tenzen-y — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**